pkg/etcd/etcd.go

```diff
@@ -1048,16 +1056,53 @@ func (e *ETCD) manageLearners(ctx context.Context) {
 			continue
 		}
 
+		nodes, err := e.getETCDNodes()
+		if err != nil {
+			logrus.Errorf("Error while listing nodes with etcd status: %v", err)
+			continue
+		}
+
+		client, err := util.GetClientSet(e.config.Runtime.KubeConfigSupervisor)
+		if err != nil {
+			logrus.Errorf("Failed to get k8s client for patch node status condition: %v", err)
+			continue
+		}
+
 		for _, member := range members.Members {
 			if member.IsLearner {
 				if err := e.trackLearnerProgress(ctx, progress, member); err != nil {
 					logrus.Errorf("Failed to track learner progress towards promotion: %v", err)
 				}
+
+				for _, node := range nodes.Items {
+					if strings.Contains(member.Name, node.Name) {
```

Comment:
I think this will cause problems - for example, if the cluster has members named `ip-172-31-4-16` and `ip-172-31-4-160`, and `ip-172-31-4-169`, the first member will be used to set the status for all three nodes. 

You should instead probably do something like this:
```suggestion
					if strings.HasPrefix(member.Name, node.Name + "-") {
```

Ref: https://github.com/k3s-io/k3s/blob/bbafb86e91ae3682a1811119d136203957df9061/pkg/etcd/etcd.go#L608